### DISCUSSION TITLE
Add option to include special character in passphrase

### DISCUSSION THREAD

**Discussion by livgm** [bitwarden/clients#6820] [Password Manager]
### Code Contribution Proposal

I think an option to include special characters in passphrases (apart from "-") would be really useful.
Many websites require a special character and "-" doesn't seem to count for a large chunk of them so I need to manually edit the passphrase afterwards.
**My idea would be to simply add a second toggle beneath the already existing "Include a number" that says "Include a special character (!&....)".**

### Thoughts:
- I am aware that one can manually edit the word separator to be a special character but that seems weird. "-" is most commonly used and the same argument could have been made for numbers ("Why include a numbers option when "apple3apple3apple" does the trick?").
- I am wondering what the preferred position within the passphrase would be. ("apple!-apple-apple" vs. "apple-!-apple")
- What special characters are useful to use in a passphrase? ("$" could be misunderstood as the word "dollar" (included in the EFF passphrase list), whereas "&" (ampersand) is not a word that can appear in a passphrase)
- Especially considering that including numbers is already an option, deviating from 'pure' passphrases, I believe including special characters shouldn't be a significant departure.
- Might make passphrases look a little too password-y? Probably not a big issue since it could just be 1 special char and should be off by default anyway.
 
All in all, I believe this is useful and cohesive with the rest of the password and passphrase options.
Happy to implement this myself and looking forward to feedback!

**Comment by dwbit** [comment]
Closing out this older request, feel free to ping if you need anything.
